File: app/src/main/java/com/osrsd/cache/def/ObjectDefinition.java
package com.osrsd.cache.def;

import lombok.Data;

import java.util.HashMap;
import java.util.Map;

@Data
public final class ObjectDefinition implements Definition {

    private int id;
    private short[] retextureToFind;
    private int decorDisplacement;
    private boolean isHollow;
    private String name;
    private int[] objectModels;
    private int[] objectTypes;
    private short[] recolorToFind;
    private int mapAreaId;
    private short[] textureToReplace;
    private int sizeX;
    private int sizeY;
    private int anInt2083;
    private int[] anIntArray2084;
    private int offsetX;
    private boolean mergeNormals;
    private int wallOrDoor;
    private int animationID;
    private int varbitID;
    private int ambient;
    private int contrast;
    private String[] actions;
    private int interactType;
    private int mapSceneID;
    private int blockingMask;
    private short[] recolorToReplace;
    private boolean shadow;
    private int modelSizeX;
    private int modelSizeHeight;
    private int modelSizeY;
    private int offsetHeight;
    private int offsetY;
    private boolean obstructsGround;
    private int contouredGround;
    private int supportsItems;
    private int[] configChangeDest;
    private boolean isRotated;
    private int varpID;
    private int ambientSoundId;
    private boolean aBool2111;
    private int anInt2112;
    private int anInt2113;
    private boolean blocksProjectile;
    private Map<Integer, Object> params;

    public ObjectDefinition(int id) {
        this.id = id;
        decorDisplacement = 16;
        isHollow = false;
        name = "null";
        mapAreaId = -1;
        sizeX = 1;
        sizeY = 1;
        anInt2083 = 0;
        offsetX = 0;
        mergeNormals = false;
        wallOrDoor = -1;
        animationID = -1;
        varbitID = -1;
        ambient = 0;
        contrast = 0;
        actions = new String[5];
        interactType = 2;
        mapSceneID = -1;
        blockingMask = 0;
        shadow = true;
        modelSizeX = 128;
        modelSizeHeight = 128;
        modelSizeY = 128;
        offsetHeight = 0;
        offsetY = 0;
        obstructsGround = false;
        contouredGround = -1;
        supportsItems = -1;
        isRotated = false;
        varpID = -1;
        ambientSoundId = -1;
        aBool2111 = false;
        anInt2112 = 0;
        anInt2113 = 0;
        blocksProjectile = true;
        params = new HashMap<>();
    }

}

File: app/src/main/kotlin/com/osedit/spring/domain/ObjectDefinition.kt
package com.osedit.spring.domain

import com.osedit.cache.Config
import com.osedit.cache.Indices
import com.osedit.cache.provider.ObjectProvider
import java.nio.ByteBuffer
import javax.persistence.Column
import javax.persistence.ElementCollection
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class ObjectDefinition(@Id private val id: Int? = 0) : Definition {
    var retextureToFind: ShortArray? = null
    var decorDisplacement = 16
    var isHollow = false
    var name: String  = "null"
    var objectModels: IntArray? = null
    var objectTypes: IntArray? = null
    var recolorToFind: ShortArray? = null
    var mapAreaId: Int = -1
    var retextureToReplace: ShortArray? = null
    var sizeX: Int = 1
    var sizeY: Int = 1
    var anInt2083: Int = 0
    var anIntArray2084: IntArray? = null
    var offsetX: Int = 0
    var mergeNormals: Boolean = false
    var wallOrDoor: Int = -1
    var animationId: Int = -1
    var varbitId: Int = -1
    var ambient: Int = 0
    var contrast: Int = 0
    @ElementCollection
    var actions = mutableListOf<String?>(null, null, null, null, null)
    var interactType: Int = 2
    var mapSceneID: Int = -1
    var blockingMask: Int = 0
    var recolorToReplace: ShortArray? = null
    var shadow: Boolean = true
    var modelSizeX: Int = 128
    var modelSizeHeight: Int = 128
    var modelSizeY: Int = 128
    var offsetHeight: Int = 0
    var offsetY: Int = 0
    var obstructsGround: Boolean = false
    var contouredGround: Int = -1
    var supportsItems: Int = -1
    @Column(length = 1055)
    var configChangeDest: IntArray? = null
    var isRotated: Boolean = false
    var varpId: Int = -1
    var ambientSoundId: Int = -1
    var aBool2111: Boolean = false
    var anInt2112: Int = 0
    var anInt2113: Int = 0
    var blocksProjectile: Boolean = true
    @ElementCollection
    var params = mutableMapOf<Int, String>()

    override fun path(): String {
        return "/objects"
    }

    override fun config(): Config {
        return Config.OBJECTS
    }

    override fun indices(): Indices {
        return Indices.CONFIG
    }

    override fun decode(byteBuffer: ByteBuffer): Definition {
        return ObjectProvider.decode(byteBuffer, this)
    }
}